pkg/util/sets/set.go

```diff
@@ -14,44 +14,49 @@
 
 package sets
 
-import "sort"
+import (
+	"golang.org/x/exp/constraints"
+	"golang.org/x/exp/slices"
+)
 
-type Set map[string]struct{}
+type Set[T constraints.Ordered] map[T]struct{}
+
+type String = Set[string]
```

Comment:
Can we please do the same for Int and also there are still usages like sets.New[string]?

Reply:
New[string] cannot be avoided in all cases, only when we have given a hint already - either by predeclaring the return type or by passing on the variadic args 

For int - I didn't think we even have any for int anymore, the one we did use I switch to uint32

Reply:
I think it's strictly worse and only feels better since generics aren't as common *yet*. Ots more wordy than just `New[string]` and less clear

Reply:
Do you have any other concerns?